src/js/stores/AppFormStore.js

```diff
@@ -92,9 +109,49 @@ function rebuildModelFromFields(app, fields, fieldId) {
   }
 }
 
+function processResponseErrors(responseErrors, response, statusCode) {
+  if (statusCode >= 500) {
+    responseErrors.general = AppFormErrorMessages.general[1];
+
+  } else if (statusCode === 422 && response != null &&
+      Util.isArray(response.errors)) {
+
+    response.errors.forEach((error) => {
+      var fieldId = error.attribute.length
+        ? error.attribute
+        : "general";
+      fieldId = responseAttributeNameToFieldIdMap[fieldId] || fieldId;
+      responseErrors[fieldId] = error.error;
+    });
+
+  } else if (statusCode === 409 && response != null &&
+      response.message != null) {
+
+    responseErrors.general =
+      `${AppFormErrorMessages.general[2]} ${response.message}`;
+
+  } else if (statusCode === 400 && response != null &&
+      Util.isArray(response.details)) {
+
+    response.details.forEach((detail) => {
+      var fieldId = detail.path.length
+        ? detail.path
+        : "general";
+      fieldId = responseAttributeNameToFieldIdMap[fieldId] || fieldId;
+      responseErrors[fieldId] = detail.errors.join(", ");
+    });
+
+  } else if (statusCode >= 300) {
```

Comment:
maybe we could refactor this with a switch, I guess readability would improve a tiny bit...